src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,204 @@
+package mate.jdbc.dao;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import mate.jdbc.exception.DataProcessingException;
+import mate.jdbc.lib.Dao;
+import mate.jdbc.model.Car;
+import mate.jdbc.model.Driver;
+import mate.jdbc.util.ConnectionUtil;
+
+@Dao
+public class CarDaoImpl implements CarDao {
+    @Override
+    public Car create(Car car) {
+        String query = "INSERT INTO cars (name, manufactur_id) VALUES (?, ?)";
+        try (Connection connection = ConnectionUtil.getConnection();
+                  PreparedStatement statement = connection.prepareStatement(query,
+                               Statement.RETURN_GENERATED_KEYS)) {
+            statement.setString(1, car.getName());
+            statement.setLong(2, car.getManufacturer().getId());
+            statement.executeUpdate();
+            ResultSet resultSet = statement.getGeneratedKeys();
+            if (resultSet.next()) {
+                car.setId(resultSet.getObject(1, Long.class));
+                List<Driver> drivers = car.getDrivers();
+                for (Driver d: drivers) {
+                    addRelation(d.getId(), car.getId());
+                }
```

Comment:
do not create a connection inside another connection. move addRelation out of try-caych